### ISSUE TITLE
make: *** [vproc_sdk] Error 2

### ISSUE THREAD

**Issue by ArrowFAE** [MicrosemiVoiceProcessing/ZLK38AVS#1]
Hello,

I just received your kit and going through the instructions.  I'm following what is written in the Wiki, however, I'm getting make errors after "sudo make all.  "Is there something missing in the make file which is causing this error?  I'm using the latest Ubuntu version.

Here's what get dumped on the terminal.

make: Circular vproc_sdk <- vproc_sdk dependency dropped.
make -C vproc_sdk vproc_sdk 
make[1]: Entering directory '/home/pi/ZLK38AVS/vproc_sdk'
Building Platform....
mkdir -p /vproc_sdk/libs
make -C /vproc_sdk/platform/raspberry/ 
make[2]: Entering directory '/home/pi/ZLK38AVS/vproc_sdk'
make[2]: *** /vproc_sdk/platform/raspberry/: No such file or directory.  Stop.
make[2]: Leaving directory '/home/pi/ZLK38AVS/vproc_sdk'
Makefile:21: recipe for target 'platform' failed
make[1]: *** [platform] Error 2
make[1]: Leaving directory '/home/pi/ZLK38AVS/vproc_sdk'
Makefile:36: recipe for target 'vproc_sdk' failed
make: *** [vproc_sdk] Error 2

Thanks in advance,
Scott

**Comment by cygnusnow** [comment]
It seems you entered a wrong directory.

**Comment by ArrowFAE** [comment]
That's what it seems, yes.  I'm executing from ~/ZLK38AVS.  When you execute the same instruction steps, what output do you see?

Scott
248-719-0792

Sorry for any typos from my mobile phone.

On Jun 14, 2017, at 5:01 AM, ruohuyang <notifications@github.com<mailto:notifications@github.com>> wrote:


It seems you entered a wrong directory.

**Comment by bonyjean1** [comment]
Hi Scott,
Basically the /vproc_sdk/platform/raspberry is missing in your downloaded ZLK38AVS repo.
I was able to reproduce the exact same error by simply deleting the raspberry folder under ZLK38AVS/vproc_sdk/platform.

You can verify that the raspberry folder is there as per below.
cd ZLK38AVS/vproc_sdk/platform

It seems the ZLK38AVS repo was not properly cloned into your pi.  

My suggestion is to download the ZLK38AVS again
git clone https://github.com/MicrosemiVoiceProcessing/ZLK38AVS

Best regards,
Jean

**Comment by ArrowFAE** [comment]
That’s the odd thing.  I have done that several times thinking the same.  However, the directory is there and still get the error.

pi@arrow:~/ZLK38AVS/vproc_sdk/platform/raspberry $ ls
docs driver include kernet Makefile readme tools

Do I need to modify my PATH?

What did you do to correct your error?

Scott
248-719-0792

**Comment by bonyjean1** [comment]
Hi Scott,
I think I see what the problem is.

You are doing a sudo make all instead of just make all. Per your log you are logged into the pi as the user pi, and have downloaded the ZLK38AVS repo not as root but as the pi user, therefore you don't sudo before the make command. Simply issue a 
make all

Best regards,
Jean

**Comment by ArrowFAE** [comment]
Thanks for pointing that out.  I’m unpacking kernel source now.  I’ll report back if something doesn’t work right.

Scott
248-719-0792

**Comment by ArrowFAE** [comment]
Oui.  More errors.

home/pi/ZLK38AVS/vproc_sdk/platform/raspberry/driver/ssl/ssl.c:366:7: warning: incompatible implicit declaration of built-in function ‘strncpy’
cc1: some warnings being treated as errors
scripts/Makefile.build:293: recipe for target '/home/pi/ZLK38AVS/vproc_sdk/platform/raspberry/driver/ssl/ssl.o' failed
make[6]: *** [/home/pi/ZLK38AVS/vproc_sdk/platform/raspberry/driver/ssl/ssl.o] Error 1
Makefile:1490: recipe for target '_module_/home/pi/ZLK38AVS/vproc_sdk/platform/raspberry/driver/ssl' failed
make[5]: *** [_module_/home/pi/ZLK38AVS/vproc_sdk/platform/raspberry/driver/ssl] Error 2
make[5]: Leaving directory '/home/pi/linux-ef3b440e0e4d9ca70060483aa33d5b1201ceceb8'
Makefile:17: recipe for target 'all' failed
make[4]: *** [all] Error 2
make[4]: Leaving directory '/home/pi/ZLK38AVS/vproc_sdk/platform/raspberry/driver/ssl'
Makefile:10: recipe for target 'ssl' failed
make[3]: *** [ssl] Error 2
make[3]: Leaving directory '/home/pi/ZLK38AVS/vproc_sdk/platform/raspberry/driver'
Makefile:10: recipe for target 'driver' failed
make[2]: *** [driver] Error 2
make[2]: Leaving directory '/home/pi/ZLK38AVS/vproc_sdk/platform/raspberry'
Makefile:21: recipe for target 'platform' failed
make[1]: *** [platform] Error 2
make[1]: Leaving directory '/home/pi/ZLK38AVS/vproc_sdk'
Makefile:36: recipe for target 'vproc_sdk' failed
make: *** [vproc_sdk] Error 2


Scott
248-719-0792

From: Scott Kerstein
Sent: Wednesday, June 14, 2017 4:09 PM
To: MicrosemiVoiceProcessing/ZLK38AVS <reply@reply.github.com>; MicrosemiVoiceProcessing/ZLK38AVS <ZLK38AVS@noreply.github.com>
Cc: Author <author@noreply.github.com>
Subject: RE: [MicrosemiVoiceProcessing/ZLK38AVS] make: *** [vproc_sdk] Error 2 (#1)

Thanks for pointing that out.  I’m unpacking kernel source now.  I’ll report back if something doesn’t work right.

Scott
248-719-0792

**Comment by bonyjean1** [comment]
Can you tell the version of :
- Your pi hardware
cat /proc/cpuinfo

- Raspian image (you can read the raspian image date/version:)
 cat /etc/rpi-issue
 cat /proc/version

- Kernel version  (you can get the kernel by issuing:)  
uname -a
or
uname -r   


Thanks,
Jean

**Comment by ArrowFAE** [comment]
I started with fresh RPi3 and Raspian download.  Thought I had updated/upgraded.  I will update again and compare.


pi@arrow:~ $ cat /proc/cpuinfo

processor : 0, 1, 2, 3

model name : ARMv7 Processor rev 4 (v7l)

BogoMIPS : 38.40

Features : half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 lpae evtstrm crc32

CPU implementer : 0x41

CPU architecture: 7

CPU variant : 0x0

CPU part : 0xd03

CPU revision : 4

Hardware : BCM2835

Revision : a02082

Serial : 00000000d99c38c6



pi@arrow:~ $ cat /etc/rpi-issue

Raspberry Pi reference 2017-04-10

Generated using pi-gen, https://github.com/RPi-Distro/pi-gen, acc4f56597e75225b93be3e1e20504a40565a9b9, stage5



pi@arrow:~ $ cat /proc/version

Linux version 4.9.24-v7+ (dc4@dc4-XPS13-9333) (gcc version 4.9.3 (crosstool-NG crosstool-ng-1.22.0-88-g8460611) ) #993 SMP Wed Apr 26 18:01:23 BST 2017



pi@arrow:~ $ uname -a

Linux arrow 4.9.24-v7+ #993 SMP Wed Apr 26 18:01:23 BST 2017 armv7l GNU/Linux
Scott
248-719-0792

**Comment by ArrowFAE** [comment]
Updated/upgraded – no change in error.

Scott
248-719-0792

From: Scott Kerstein
Sent: Wednesday, June 14, 2017 6:06 PM
To: MicrosemiVoiceProcessing/ZLK38AVS <reply@reply.github.com>; MicrosemiVoiceProcessing/ZLK38AVS <ZLK38AVS@noreply.github.com>
Cc: Author <author@noreply.github.com>
Subject: RE: [MicrosemiVoiceProcessing/ZLK38AVS] make: *** [vproc_sdk] Error 2 (#1)

I started with fresh RPi3 and Raspian download.  Thought I had updated/upgraded.  I will update again and compare.


pi@arrow:~ $ cat /proc/cpuinfo

processor : 0, 1, 2, 3

model name : ARMv7 Processor rev 4 (v7l)

BogoMIPS : 38.40

Features : half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 lpae evtstrm crc32

CPU implementer : 0x41

CPU architecture: 7

CPU variant : 0x0

CPU part : 0xd03

CPU revision : 4

Hardware : BCM2835

Revision : a02082

Serial : 00000000d99c38c6



pi@arrow:~ $ cat /etc/rpi-issue

Raspberry Pi reference 2017-04-10

Generated using pi-gen, https://github.com/RPi-Distro/pi-gen, acc4f56597e75225b93be3e1e20504a40565a9b9, stage5



pi@arrow:~ $ cat /proc/version

Linux version 4.9.24-v7+ (dc4@dc4-XPS13-9333) (gcc version 4.9.3 (crosstool-NG crosstool-ng-1.22.0-88-g8460611) ) #993 SMP Wed Apr 26 18:01:23 BST 2017



pi@arrow:~ $ uname -a

Linux arrow 4.9.24-v7+ #993 SMP Wed Apr 26 18:01:23 BST 2017 armv7l GNU/Linux
Scott
248-719-0792

**Comment by bonyjean1** [comment]
The kernel on your pi is 4.9.24, that's is an experimental kernel, I wouldn't use the ZLK380AVS on an experimnetal kernel. You are using Raspian distro 2017-04-10 which by default includes kernel 4.4.50-v7+ which is a stable kernel. Have you manually upgraded the kernel?

The sequence to porting our ZLK380AVS on a pi is the following 3 steps below:
- Download the latest Raspian image install that image on an SD card, and boot your pi with that SD card
- Download the ZLK380AVS from github into the pi (Make sure you have the PI connected to internet)
- cd into the ZLK380AVS and do a 
make all

The make all process will update the pi with all the necessary packages needed for its compilation and installation. 
-----------------------------------------------------------
Below are the same info on my system:

pi@JBraspberrypi3:~/AVS/ZLK38AVS $ cat /etc/rpi-issue
Raspberry Pi reference 2017-04-10
Generated using pi-gen, https://github.com/RPi-Distro/pi-gen, acc4f56597e75225b93be3e1e20504a40565a9b9, stage5

pi@JBraspberrypi3:~/AVS/ZLK38AVS $ cat /proc/version
Linux version 4.4.50-v7+ (dc4@dc4-XPS13-9333) (gcc version 4.9.3 (crosstool-NG crosstool-ng-1.22.0-88-g8460611) ) #970 SMP Mon Feb 20 19:18:29 GMT 2017

pi@JBraspberrypi3:~/AVS/ZLK38AVS $ cat /proc/cpuinfo
processor	: 0
model name	: ARMv7 Processor rev 4 (v7l)
BogoMIPS	: 76.80
Features	: half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 lpae evtstrm crc32 
CPU implementer	: 0x41
CPU architecture: 7
CPU variant	: 0x0
CPU part	: 0xd03
CPU revision	: 4
processor	: 1
model name	: ARMv7 Processor rev 4 (v7l)
BogoMIPS	: 76.80
Features	: half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 lpae evtstrm crc32 
CPU implementer	: 0x41
CPU architecture: 7
CPU variant	: 0x0
CPU part	: 0xd03
CPU revision	: 4
processor	: 2
model name	: ARMv7 Processor rev 4 (v7l)
BogoMIPS	: 76.80
Features	: half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 lpae evtstrm crc32 
CPU implementer	: 0x41
CPU architecture: 7
CPU variant	: 0x0
CPU part	: 0xd03
CPU revision	: 4
processor	: 3
model name	: ARMv7 Processor rev 4 (v7l)
BogoMIPS	: 76.80
Features	: half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 lpae evtstrm crc32 
CPU implementer	: 0x41
CPU architecture: 7
CPU variant	: 0x0
CPU part	: 0xd03
CPU revision	: 4
Hardware	: BCM2709
Revision	: a22082
Serial		: 00000000f4fe4aeb

pi@JBraspberrypi3:~/AVS/ZLK38AVS $ gcc --version
gcc (Raspbian 4.9.2-10) 4.9.2
Copyright (C) 2014 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

pi@JBraspberrypi3:~/AVS/ZLK38AVS $ uname -a
Linux JBraspberrypi3 4.4.50-v7+ #970 SMP Mon Feb 20 19:18:29 GMT 2017 armv7l GNU/Linux

**Comment by ArrowFAE** [comment]
I downloaded the latest Raspian from the RP website.
https://www.raspberrypi.org/downloads/raspbian/

After I download and image to an SD card, I performed update/upgrade commands.

Are you suggesting that I just use the Raspian 4-10 image, as is without updating/upgrading?  If so, this should be documented somehow for our customer use to avoid similar issues and make it a good out-of-box experience.

Suggestion might be to offer a download link to a compressed working AVS image, then use Win32 Imager tool, or other tool, to prepare the SD card.  Not sure how to manage existing RP3 users that may have upgraded already for other purposes with their Pi.  Switching cards isn’t a bad idea, just cumbersome.

I’ll re-image using 4-10 and compare to your system info below and report back.

Scott
248-719-0792

**Comment by bonyjean1** [comment]
Hi Scott,
Do not do the upgrade command. That command will simply upgrade to the latest greatest available kernel and packages regardless whether the kernel and packages are experimental or not.

Once you flash your pi with the latest Raspian, your sequence must be.

- Download the ZLK380AVS from github into the pi (Make sure you have the PI connected to internet)
git clone https://github.com/MicrosemiVoiceProcessing/ZLK38AVS

- cd into the ZLK380AVS and do a
make all

**Comment by bonyjean1** [comment]
Hi Scott,
The steps are documented in Chapter 5 of the ZLK380AVS/docs/Microsemi_ZLK38AVS_User_Guide.pdf

Once you have your pi flashed with the latest Raspian image, your pi is ready for ZLK380AVS installation.
The make all will do everything needed (including searching for required/appropriate packages,  you need an internet connection to the pi throughout the ZLK380AVS compilation process) to make sure compilation and installation are successful.

Best regards,
Jean

**Comment by ArrowFAE** [comment]
Thank you, I got it working now.  Thank you for your help.

Neat demo and it seems to work better than the three Alexa’s in my house….

From reading the Users Manual, I didn’t realize that a certain kernel was required.  I have a couple RPi3s laying around and wanted to get something up quickly for a customer without having to re-image and wipe out other things that I’m doing with the RPi3.

It might be helpful if you could document in the Users Guide that it’s important that one doesn’t upgrade the kernel and stick with the 4-10 kernel.  One question, when the kernel likely updates in October without experimental changes, will this demo still work?

Another possible issue that I observed during the Amazon ID setup.  Of I were to answer “n” to the questions asking if what I entered was okay and then proceeded to enter the correct information, the echo back from what I entered didn’t appear.  What appeared was the old entry and in the wrong order of questions (Client ID and Client Secret.)  I had to start over to fix things.

Scott
248-719-0792

**Comment by bonyjean1** [comment]
Hi Scott,
We basically keep track of which stable kernels are available for the pi and check our codes against these kernels.
These kernels and firmware for the pi are available here.
https://github.com/Hexxeh/rpi-firmware/commits/master?after=9def95edfc121e3762a9ff520befbee9557436e1+34

So far the latest stable kernel available at the above link for the pi is 4.4.50, and the latest experimental/development kernel is 4.9.x. We normally don't use the experimental kernels. So far we have tried our code on past stable 3.x and 4.x kernels and never found an issue.

Our ZLK380AVS is not restricted to kernel 4.4.50, as mentioned above  we have tested this againts past stable kernel with no issue.

We are keeping track with stable kernel release, and will update the kit accordingly.

Best regards,
Jean
